packages/map-template/src/components/Map/MapboxMap/MapboxMap.jsx

```diff
@@ -64,6 +73,40 @@ function MapboxMap({ onMapView, onPositionControl }) {
     }, []); // eslint-disable-line react-hooks/exhaustive-deps
     // We ignore eslint warnings about missing dependencies because onMapView should never change runtime and changing Mapbox Access Token runtime will give other problems.
 
+    // If the map is ready, check if the 3D features are toggled 
+    useEffect(() => {
+        if (mapView && mapView.isReady && isMapReady) {
+            // Check if the 3D features are switched on
+            // Conditionally render either 2D or 3D features depending on the value of is3DToggled
+            // Tilt the map to 45 degrees when the 3D features are switched on, and to 0 degrees when the 2D features are switched on.
+            if (is3DToggled) {
+                mapView.hideFeatures([mapView.MapboxFeatures.MODEL2D, mapView.MapboxFeatures.WALLS2D])
+                mapView.tilt(45, 2000);
```

Comment:
Does this collide with the `pitch` property? Or the `timeout` property?

Reply:
Good find @ammapspeople 
Pushed the new changes with handling the initial value of the pitch and the timeout property.